### PR TITLE
Longhaul: Plumbing and tolerance adjustments

### PR DESCRIPTION
**Pull request by and-rewsmith**
I split this into two PRs. The first (#5902) has been merged. The second is #5916 .

### CODE REVIEW

File: builds/e2e/connectivity.yaml [hunk 1]
```diff
@@ -218,6 +218,7 @@ jobs:
           customEdgeHub.image: '$(customEdgeHub.image)'
           test.runtimeLogLevel: '$(test.runtimeLogLevel)'
           testInfo.testName: '$(testInfo.testName)'
+          topology: 'SingleNode'
```
Comment (and-rewsmith) [inline builds/e2e/connectivity.yaml:221]: We need to inject a `topology` var from yaml->script only for single node connectivity since it shares a deployment file across single-node/nested. For nested connectivity and all longhaul pipelines, deployment files are not shared so the value can be hardcoded.

File: test/modules/TwinTester/StatusCode.cs [hunk 2]
```diff
@@ -14,6 +14,6 @@ enum StatusCode
         ReportedPropertyUpdateCallFailure = 504,
         ReportedPropertyUpdateNotInCloudTwin = 505,
         ReportedPropertyUpdateNotInCloudTwinAfterCleanUpThreshold = 506,
-        ReportedPropertyReceivedAfterThreshold = 507,
+        ReportedPropertyReceivedAfterThreshold = 205,
```
Comment (and-rewsmith) [inline test/modules/TwinTester/StatusCode.cs:17]: Changing this to be a 2xx status code as it indicates success. This will make the tolerance logic cleaner.

File: test/modules/TestResultCoordinator/Reports/LegacyTwin/LegacyTwinReportGenerator.cs [hunk 3]
```diff
@@ -70,5 +75,82 @@ public async Task<ITestResultReport> CreateReportAsync()
             Logger.LogInformation($"Successfully finished creating LegacyTwinReport for Source [{this.SenderSource}]");
             return report;
         }
+
+        // See TwinTester/StatusCode.cs for reference.
+        bool IsPassed(IDictionary<int, int> results)
```
Comment (nimanch) [inline test/modules/TestResultCoordinator/Reports/LegacyTwin/LegacyTwinReportGenerator.cs:80]: Recommend renaming
```suggestion
        bool IsPassed(IDictionary<int, int> statusCodesToCount)
```

**Comment by varunpuranik** [comment]
The changes to plumb through additional information and flags is great. 
But the key changes are to the test themselves and adjusting the tolerance - those might end up hiding underlying issues in the product, and need to be carefully considered. And if needed, we need to have PBIs to revert the changes after fixing underlying issues. Going forward, can we separate out these changes from the plumbing? It is hard to follow that in large PRs like these. 
For this PR - can you please add details on what changes you made, related to tolerance adjustments?

**Comment by and-rewsmith** [comment]
@varunpuranik Didn't expect as many changes from the plumbing as there were. I agree they are distracting. It was easy enough to isolate the plumbing into its own diff, so I have this draft PR here:
https://github.com/Azure/iotedge/pull/5902

I'm waiting on test pass before opening #5902 up for review. Then we can handle the tolerance specific stuff, which I've cleaned up and documented in this PR separately. Converting this PR to draft.

**Comment by and-rewsmith** [comment]
Closing in favor of #5916 for cleanup.
